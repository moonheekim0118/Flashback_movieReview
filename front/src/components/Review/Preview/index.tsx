import React , { useCallback } from 'react';
import { useDispatch } from 'react-redux';
import Router from 'next/router';
import styled from 'styled-components';
import Badge from '../Badge';
import Icon from '../../../atoms/Icons';
import ConfirmAlert from '../../ConfirmAlert';
import useAlert from '../../../hooks/useAlert';
import { removeMyReviewAction } from '../../../actions/review';
import { faTrash } from '@fortawesome/free-solid-svg-icons';
import { ReviewList } from '../../../model/ReviewList';
import { Container, 
         MoviePoster,
         MovieDescription 
} from '../../Movie/MovieCard';

interface Props {
    Review:ReviewList; 
}

// 영화 리뷰 미리보기 컴포넌트 
const Preview =({Review}:Props)=>{
    const dispatch = useDispatch();
    
    const [ showConfirmAlert, 
            openConfirmAlert, 
            closeConfirmAlert 
        ] = useAlert();
    
    // 프리뷰 클릭시 해당 리뷰보기 창으로 리다이렉트 
    const onMove = useCallback(()=>{
        Router.push(`/singleReview/${Review.id}`);
    },[]);

    //리뷰 삭제 아이콘 클릭시, confirmAlert 띄워줌 
    const openRemoveAlert = useCallback((e)=>{ // alert 띄워주기 
        e.stopPropagation();
        openConfirmAlert();
    },[]);

    // confirmAlert 닫아줌 
    const closeRemoveAlert = useCallback((e)=>{ // alert 닫아주기 
        e.stopPropagation();
        closeConfirmAlert();
    },[]);

    // 리뷰 삭제 
    const onClickRemove = useCallback((e)=>{ // 리뷰 삭제 
        e.stopPropagation(); 
        dispatch(removeMyReviewAction(Review.id));
        closeConfirmAlert(); // confirmAlert 닫아줌 
    },[]);

    return(
        <Container onClick={onMove}>
            {showConfirmAlert && 
            <ConfirmAlert 
            text="정말 삭제하시겠습니까?"
            clickYes={onClickRemove} 
            clickNo={closeRemoveAlert}/>}
            <MoviePoster src={Review.Movie.image}/>
            <MovieDescription>
                <Comment>{Review.shortComment}</Comment>
                <Badge badgeName={Review.rating} selected={true}/>
            </MovieDescription>
            <Icon icon={faTrash} className="faTrash" onClick={openRemoveAlert}/>
        </Container>
    );
}

const Comment = styled.div`
    font-size:1.5rem;
    font-weight:bold;
`;

export default Preview;