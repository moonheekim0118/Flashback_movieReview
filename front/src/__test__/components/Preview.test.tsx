import 'jsdom-global/register'; 
import React from "react";
import { ReviewProp } from './mock.data';
import { mount } from 'enzyme';
import Preview from '../../components/Review/Preview';

jest.mock('react-redux', ()=>({
    useDispatch : jest.fn(),
}));



describe('<Preview/>',()=>{
    let container;
    beforeEach(()=>{
        container = mount( <Preview Review={ReviewProp}/>  )
    })

    it('should match correctly', ()=>{
        expect(container.html()).toMatchSnapshot();
    })
});