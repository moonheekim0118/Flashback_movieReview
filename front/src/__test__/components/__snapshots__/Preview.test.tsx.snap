// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Preview/> should match correctly 1`] = `"<div class=\\"MovieCard__Container-sc-1v9tm0k-0 hPqAwq\\"><img src=\\"test\\" class=\\"MovieCard__MoviePoster-sc-1v9tm0k-1 hGSCZO\\"><div class=\\"MovieCard__MovieDescription-sc-1v9tm0k-2 jyXyWd\\"><div class=\\"Preview__Comment-sc-2oahz8-0 buRmkP\\">test</div><div class=\\"Badge__Container-sc-1qprtwz-0 kPfNfx\\"><svg aria-hidden=\\"true\\" focusable=\\"false\\" data-prefix=\\"fas\\" data-icon=\\"thumbs-up\\" class=\\"svg-inline--fa fa-thumbs-up fa-w-16 Icons__StyledIcon-khcp93-0 bOobqK GOOD\\" role=\\"img\\" xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 512 512\\" iconsize=\\"24\\" color=\\"purple\\"><path fill=\\"currentColor\\" d=\\"M104 224H24c-13.255 0-24 10.745-24 24v240c0 13.255 10.745 24 24 24h80c13.255 0 24-10.745 24-24V248c0-13.255-10.745-24-24-24zM64 472c-13.255 0-24-10.745-24-24s10.745-24 24-24 24 10.745 24 24-10.745 24-24 24zM384 81.452c0 42.416-25.97 66.208-33.277 94.548h101.723c33.397 0 59.397 27.746 59.553 58.098.084 17.938-7.546 37.249-19.439 49.197l-.11.11c9.836 23.337 8.237 56.037-9.308 79.469 8.681 25.895-.069 57.704-16.382 74.757 4.298 17.598 2.244 32.575-6.148 44.632C440.202 511.587 389.616 512 346.839 512l-2.845-.001c-48.287-.017-87.806-17.598-119.56-31.725-15.957-7.099-36.821-15.887-52.651-16.178-6.54-.12-11.783-5.457-11.783-11.998v-213.77c0-3.2 1.282-6.271 3.558-8.521 39.614-39.144 56.648-80.587 89.117-113.111 14.804-14.832 20.188-37.236 25.393-58.902C282.515 39.293 291.817 0 312 0c24 0 72 8 72 81.452z\\"></path></svg>좋아요!</div></div><svg aria-hidden=\\"true\\" focusable=\\"false\\" data-prefix=\\"fas\\" data-icon=\\"trash\\" class=\\"svg-inline--fa fa-trash fa-w-14 Icons__StyledIcon-khcp93-0 bOobqK faTrash\\" role=\\"img\\" xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 448 512\\" iconsize=\\"24\\" color=\\"purple\\"><path fill=\\"currentColor\\" d=\\"M432 32H312l-9.4-18.7A24 24 0 0 0 281.1 0H166.8a23.72 23.72 0 0 0-21.4 13.3L136 32H16A16 16 0 0 0 0 48v32a16 16 0 0 0 16 16h416a16 16 0 0 0 16-16V48a16 16 0 0 0-16-16zM53.2 467a48 48 0 0 0 47.9 45h245.8a48 48 0 0 0 47.9-45L416 128H32z\\"></path></svg></div>"`;
